app/models/concerns/foreman_ansible/host_managed_extensions.rb

```diff
@@ -15,12 +15,11 @@ module HostManagedExtensions
                     :only_explicit => true
 
       before_provision :play_ansible_roles
-      include ForemanAnsible::HasManyAnsibleRoles
       audit_associations :ansible_roles
 
       def inherited_ansible_roles
         return [] unless hostgroup
-        hostgroup.all_ansible_roles
+        hostgroup.inherited_and_own_ansible_roles
```

Comment:
Isn't this the only thing that changed? As far as I can understand from the changes, the module `HasManyAnsibleRoles` was doing the same exactly that this is doing:

`(ansible_roles + inherited_ansible_roles + host_ansible_roles).uniq` for Hostgroup
`(ansible_roles + inherited_ansible_roles).uniq` for Host

Reply:
yes, but if I kept it in all_ansible_roles method, I'd need to add additional argument whether when I ask for host roles I want to include hostgroups' hosts and their role or not. That was second if in the method that would need to check for context where it's called, that is a good indication that this DRYification was not a good idea, since these two methods are different.